SC_redesign/Crop_and_Soil/field/field.py

```diff
@@ -126,31 +127,131 @@ def _composition_sums_to_one(self) -> bool:
+        passing those amount to the FertilizerApplication module, and recording the fertilizer application to the
+        OutputManager.
 
-    def amend_soil(self) -> None:
-        """amend the soil with nutrients"""
-        self.soil.phosphorus_cycling.fertilizer.add_fertilizer_phosphorus(0)
-        return
+        """
+        try:
+            fertilizer_mix = self.available_fertilizer_mixes.get(mix_name)
+        except KeyError:
+            raise KeyError(f"'{self.field_data.name}': expected to have fertilizer mix for '{mix_name}', "
+                           f"received '{self.available_fertilizer_mixes}'.")
+        nitrogen_fraction = fertilizer_mix.get("N")
+        phosphorus_fraction = fertilizer_mix.get("P")
+        potassium_fraction = fertilizer_mix.get("K")
+
+        fertilizer_applied = self._formulate_fertilizer_required(nitrogen_fraction, phosphorus_fraction,
+                                                                 potassium_fraction, requested_nitrogen,
+                                                                 requested_phosphorus)
+        total_mass_applied = fertilizer_applied.get("mass")
+        phosphorus_applied = fertilizer_applied.get("phosphorus_mass")
+        nitrogen_applied = fertilizer_applied.get("nitrogen_mass")
+        potassium_applied = fertilizer_applied.get("potassium_mass")
+
+        # TODO: specify these fractions in fertilizer mixes - issue #573
+        inorganic_nitrogen_fraction = nitrogen_applied / total_mass_applied
+        ammonium_fraction = 0.0
+        organic_nitrogen_fraction = 0.0
+
+        self.fertilizer_applicator.apply_fertilizer(phosphorus_applied, total_mass_applied, inorganic_nitrogen_fraction,
+                                                    ammonium_fraction, organic_nitrogen_fraction,
+                                                    self.field_data.field_size)
+
+        self._record_fertilizer_application(mix_name, total_mass_applied, nitrogen_applied, phosphorus_applied,
+                                            potassium_applied, year, day)
+
+    @staticmethod
+    def _formulate_fertilizer_required(nitrogen_fraction: float, phosphorus_fraction: float,
+                                       potassium_fraction: float, requested_nitrogen: float,
+                                       requested_phosphorus: float) -> Dict[str, float]:
+        """
+        Determines the total mass of a specific fertilizer mix needed to meet the specified nutrient requirements.
+
+        Parameters
+        ----------
+        nitrogen_fraction : float
+            Fraction of fertilizer mix that is nitrogen, in range [0.0, 1.0]
+        phosphorus_fraction : float
+            Fraction of fertilizer mix that is phosphorus, in range [0.0, 1.0]
+        potassium_fraction : float
+            Fraction of fertilizer mix that is potassium, in range [0.0, 1.0]
```

Comment:
```suggestion
        nitrogen_fraction : float
            Fraction of fertilizer mix that is nitrogen, in range [0.0, 1.0] (unitless)
        phosphorus_fraction : float
            Fraction of fertilizer mix that is phosphorus, in range [0.0, 1.0] (unitless)
        potassium_fraction : float
            Fraction of fertilizer mix that is potassium, in range [0.0, 1.0] (unitless)
```
Just a suggestion to maintain unit labeling as other refactored files.